specs/Print.md

```diff
@@ -0,0 +1,816 @@
+                defaultInput: "");
+  if (dialog.ShowDialog() == true)
+  {
+    printerName = dialog.Input.Text;
+  }
+  return printerName;
+
+  // or
+  //
+  // Use GetPrintQueues() of LocalPrintServer from System.Printing to get list of locally installed printers.
+  // Display the printer list to the user and get the desired printer to print.
+  // Return the user selected printer name.
+}
+
+// Function to get print settings for the selected printer.
+// You may also choose get the capabilties from the native printer API, display to the user to get
+ // the print settings for the current web page and for the selected printer.
+CoreWebView2PrintSettings GetSelectedPrinterPrintSettings(string printerName)
+{
+  CoreWebView2PrintSettings printSettings = null;
+  printSettings = WebViewEnvironment.CreatePrintSettings();
+  printSettings.ShouldPrintBackgrounds = true;
+  printSettings.ShouldPrintHeaderAndFooter = true;
+
+  return printSettings;
+
+  // or
+  // Get PrintQueue for the selected printer and use GetPrintCapabilities() of PrintQueue from System.Printing
+  // to get the capabilities of the selected printer.
+  // Display the printer capabilities to the user along with the page settings.
+  // Return the user selected settings.
+}
+```
+
+## PrintToPdfStream
+
+You can use `PrintToPdfStream` method to display as a preview in a custom print dialog.
+The app can also use this API to configure the custom print preview dialog unlike confining
+print preview dialog to the webview2 control.
+
+```cpp
+// This example prints the Pdf data of the current web page to a stream.
+
+// Function to display current web page pdf data in a custom print preview dialog.
+static void DisplayPdfDataInPrintDialog(IStream* pdfData)
+{
+  // You can display the printable pdf data in a custom print preview dialog to the end user.
+}
+
+bool AppWindow::PrintToPdfStream()
+{
+  wil::com_ptr<ICoreWebView2_15> webView2_15;
+  CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+
+  wil::com_ptr<ICoreWebView2Environment6> webviewEnvironment6;
+  CHECK_FAILURE(m_webViewEnvironment->QueryInterface(IID_PPV_ARGS(&webviewEnvironment6)));
+
+  wil::com_ptr<ICoreWebView2PrintSettings> printSettings = nullptr;
+  CHECK_FAILURE(webviewEnvironment6->CreatePrintSettings(&printSettings));
+
+  wil::com_ptr<ICoreWebView2PrintSettings2> printSettings2;
+  CHECK_FAILURE(printSettings->QueryInterface(IID_PPV_ARGS(&printSettings2)));
+
+  CHECK_FAILURE(printSettings2->put_ShouldPrintBackgrounds(true));
+
+  wil::unique_cotaskmem_string title;
+  CHECK_FAILURE(m_webView->get_DocumentTitle(&title));
+
+  CHECK_FAILURE(
+      webView2_15->PrintToPdfStream(
+          printSettings2.get(),
+          Callback<ICoreWebView2PrintToPdfStreamCompletedHandler>(
+              [title = title.get(), this](HRESULT errorCode, IStream* pdfData) -> HRESULT
+              {
+                DisplayPdfDataInPrintDialog(pdfData);
```

Comment:
This assumes that the received stream has been rewound to the start of the PDF data. Is that a safe assumption?

Reply:
The stream is beginning of the document.

Reply:
Please document that the stream is rewound to the start.